internal/dag/dag.go

```diff
@@ -409,6 +409,9 @@ type Cluster struct {
 
 	// ResponseHeadersPolicy defines how headers are managed during forwarding
 	ResponseHeadersPolicy *HeadersPolicy
+
+	// SNI describes how the SNI is set on a Cluster
+	SNI string
```

Comment:
I think this could use a little explanation of how the field works.

What happens when it's set?

What happens when it's the empty string?

It seems from the rest of the code that if it's not set having the field be present does nothing, it would be good to have that contract explicitly stated here.

A link to the Envoy docs or something would maybe do as well.

Reply:
That's great, thanks.